### PR TITLE
Test/google map helper test

### PR DESCRIPTION
**Pull request by Mathematou**
This PR add all the tests for GoogleMapHelper to a level of 96%, the rest of them need the context which cannot be tested because google map cannot be instanciated and tested on cirrius.

There is also the possibility to modify an area after creating it

Thank you for your reviews!

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -34,6 +34,9 @@ android {
     kotlinOptions {
         jvmTarget = '1.8'
     }
+    //testOptions {
```
Comment (Mathematou) [inline app/build.gradle:37]: I will delete this Right now, totally forgot about this

File: app/src/test/java/com/github/sdpteam15/polyevents/GoogleMapHelperTest.kt [hunk 2]
```diff
@@ -0,0 +1,508 @@
+package com.github.sdpteam15.polyevents
```
Comment (cpittet) [inline app/src/test/java/com/github/sdpteam15/polyevents/GoogleMapHelperTest.kt:1]: :+1: :+1:

File: app/src/test/java/com/github/sdpteam15/polyevents/MockMatcher.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
+package com.github.sdpteam15.polyevents
```
Comment (cpittet) [inline app/src/test/java/com/github/sdpteam15/polyevents/MockMatcher.kt:1]: What is the purpose of this file ? I'm not sure I get it

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 4]
```diff
@@ -100,6 +104,8 @@ object GoogleMapHelper {
             areasPoints.clear()
             //Draw all areas and points
             for ((k, v) in temp) {
+                println("second : ${v.second}")
+                println("first : ${v.first}")
```
[the remark this answers is not in the document]
Comment (Rygone) [reply app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:108]: yes, remove is the good choice

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 5]
```diff
@@ -228,79 +233,95 @@ object GoogleMapHelper {
...
-        )
+
+        moveMarker = map!!.addMarker(newMarker(posCenter, 0.5f, 0.5f, PolygonAction.MOVE.toString(), null, true, R.drawable.ic_move, 0, 0, 100, 100, 100, 100))
         movePos = moveMarker!!.position
     }
 
+    /**
+     * Sets the values for the markers
+     */
+    fun newMarker(pos: LatLng, hAnchor: Float, vAnchor: Float, snippet: String?, title: String?, draggable: Boolean, idDrawable: Int, leftBound: Int, topBound: Int, rightBound: Int, bottomBound: Int, width: Int, height: Int): MarkerOptions {
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:307]: there are a lot of parameter to this method

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 6]
```diff
@@ -344,25 +376,22 @@ object GoogleMapHelper {
...
 
+        val t1 = pos.position.latitude
+        val t2 = moveDiagPos!!.latitude
+
         //Vector of the marker
         val vec = LatLng(
-            pos.position.latitude - moveDiagPos!!.latitude,
-            pos.position.longitude - moveDiagPos!!.longitude
+                t1 - t2,
+                pos.position.longitude - moveDiagPos!!.longitude
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:385]: t1 and t2 are only use ones

File: app/src/test/java/com/github/sdpteam15/polyevents/MockMatcher.kt (hunk 3 above)
Comment (Rygone) [reply app/src/test/java/com/github/sdpteam15/polyevents/MockMatcher.kt:1]: get from master PR #142

**Review by Rygone — APPROVED** [review]

**Review by cpittet — APPROVED** [review]

**Pull request merged**